apps/ocpp/tests/test_view_common.py

```
@@ -38,3 +70,60 @@ def test_charger_last_seen_handles_attribute_error(monkeypatch):
+            reference_time=reference_time,
+            active_delta=active_delta,
+        )
+        is True
+    )
+
+
+def test_charger_state_uses_recent_heartbeat_when_disconnected(monkeypatch, settings):
+    settings.NODE_LAST_SEEN_ACTIVE_DELTA = datetime.timedelta(minutes=5)
+    recent = timezone.now() - datetime.timedelta(minutes=1)
+    charger = _ChargerStub(last_heartbeat=recent)
+    monkeypatch.setattr(common.store, "is_connected", lambda *args, **kwargs: False)
```

Ruff (0.14.14)

[warning] Unused lambda argument: `args`

(ARG005)